pnc-rest/src/main/java/org/jboss/pnc/rest/endpoint/ProjectEndpoint.java

```diff
@@ -59,8 +62,10 @@ public Response createNew(@NotNull @Valid ProjectRest projectRest, @Context UriI
     @ApiOperation(value = "Updates an existing Project")
     @PUT
     @Path("/{id}")
-    public Response update(@ApiParam(value = "Project id", required = true) @PathParam("id") Integer productId,
+    public Response update(@ApiParam(value = "Project id", required = true) @PathParam("id") Integer id,
             @NotNull @Valid ProjectRest projectRest, @Context UriInfo uriInfo) {
+        validateIdIsNull(projectRest);
```

Comment:
And some another idea - Couldn't we use Validation API here and create some annotation? The entity is already processed by Validation api, so I think - all we need to do is to create just another validation annotation and use it here[1].

[1] https://docs.jboss.org/hibernate/validator/4.0.1/reference/en/html/validator-customconstraints.html